### PR TITLE
bpftop: init at 0.2.1

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/bp/bpftop/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  cargoHash = "sha256-+zh7GZ/fbhxLNQkkHFZqtJxy2IeS+KX5s2Qi5N21u/0=";
+
+  buildInputs = [ elfutils zlib libbpf ];
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  meta = with lib; {
+    description = "Bpftop provides a dynamic real-time view of running eBPF programs. It displays the average runtime, events per second, and estimated total CPU % for each program";
```
Comment (oluceps) [inline pkgs/by-name/bp/bpftop/package.nix:29]: The description may be shorter.

File: pkgs/by-name/bp/bpftop/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    pkg-config
+  ];
+
+  meta = with lib; {
+    description = "Bpftop provides a dynamic real-time view of running eBPF programs";
+    homepage = "https://github.com/Netflix/bpftop";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ mfrw ];
+    mainProgram = "bpftop";
+  };
```
Comment (jnsgruk) [inline pkgs/by-name/bp/bpftop/package.nix:34]:
```suggestion
  meta = {
    description = "Bpftop provides a dynamic real-time view of running eBPF programs";
    homepage = "https://github.com/Netflix/bpftop";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ mfrw ];
    mainProgram = "bpftop";
  };
```

File: pkgs/by-name/bp/bpftop/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "v${version}";
+    hash = "sha256-HP8ubzCfBNgISrAyLACylH4PHxLhJPzIQFmIWEL5gjo=";
+  };
+
+  cargoHash = "sha256-+zh7GZ/fbhxLNQkkHFZqtJxy2IeS+KX5s2Qi5N21u/0=";
+
+  buildInputs = [ elfutils zlib libbpf ];
+  nativeBuildInputs = [
+    pkg-config
+  ];
```
Comment (jnsgruk) [inline pkgs/by-name/bp/bpftop/package.nix:26]:
```suggestion
  buildInputs = [ 
    elfutils
    libbpf
    zlib  
  ];
  
  nativeBuildInputs = [
    pkg-config
  ];
```

Comment (jnsgruk) [reply pkgs/by-name/bp/bpftop/package.nix:26]: Minor nit to make diffs a little easier in the future

**Comment by jnsgruk** [comment]
Result of `nixpkgs-review pr 291722` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bpftop</li>
  </ul>
</details>

File: pkgs/by-name/bp/bpftop/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    libbpf
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  meta = {
+    description = "Bpftop provides a dynamic real-time view of running eBPF programs";
```
Comment (jnsgruk) [inline pkgs/by-name/bp/bpftop/package.nix:34]:
```suggestion
    description = "A dynamic real-time view of running eBPF programs";
```

Comment (jnsgruk) [reply pkgs/by-name/bp/bpftop/package.nix:34]: Sorry I missed this the first time around - the description shouldn't refer to the package name (see [docs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes)).

Good to merge once this is addressed.

**Review by jnsgruk — APPROVED** [review]
Thank you!

**Pull request merged**